src/test/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImplTest.java

```diff
@@ -228,5 +230,66 @@ public void testFromProjectWithDockerAlfrescoPlugin() {
         assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("SUB_PROJECT1_DOCKER_IMAGE"));
     }
 
+    @Test
+    public void testFromProjectWithEnvironmentVariable(){
+        composeConvention.fromProject("testname",project);
+        TaskCollection<DockerBuildImage> buildImageTaskCollection = project.getTasks().withType(DockerBuildImage.class);
+        assertTaskDependsOn(composeSettings.getUpTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getBuildTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getPushTask(), dockerBuildImageProvider);
+
+        runSetupImageEnvironmentWithComposeUp();
+
+        assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("TESTNAME_BUILD_DOCKER_IMAGE"));
+    }
+
+    @Test
+    public void testFromProjectWithEnvironmentVariable1(){
+        TaskProvider<DockerBuildImage> dockerBuildImageProvider = project.getTasks()
+                .register("buildDockerImageAlternative", DockerBuildImage.class);
+        dockerBuildImageProvider.configure(dockerBuildImage -> {
+            dockerBuildImage.getImageId().set(DOCKER_IMAGE_ID);
+        });
+        composeConvention.fromProject("testname",project);
+        TaskCollection<DockerBuildImage> buildImageTaskCollection = project.getTasks().withType(DockerBuildImage.class);
+        assertTaskDependsOn(composeSettings.getUpTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getBuildTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getPushTask(), dockerBuildImageProvider);
+
+        runSetupImageEnvironmentWithComposeUp();
+
+        assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("TESTNAME_BUILD_DOCKER_IMAGE"));
+        assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("TESTNAME_BUILD_DOCKER_IMAGE_ALTERNATIVE"));
+    }
+
+    @Test
+    public void testFromOtherProjectWithEnvironmentVariable() {
+        Project subProject1 = ProjectBuilder.builder().withParent(project).withName("sub-project1").build();
+        Project subProject2 = ProjectBuilder.builder().withParent(project).withName("sub-project2").build();
```

Comment:
Testing with just one subproject is also fine, it's not necessary to do it with 2 subprojects.